### PR TITLE
Add support for SDBus interfaces to launch and stop batch server

### LINKED ISSUE #1988: Added methods to support start_batch() and stop_batch() to SDBus and SDBusMessage C++ class
> estimate 3

Parent story: #1962

Done criteria:

- [x] Add support for appending geopm_request_s structures to a SDBusMessage
- [x] Add support to SDBus for calling a method that takes one integer.
- [x] Unit tests added for new class methods except: non-test
      constructors, private methods or classes that provide a
      last-level-abstraction of an external dependency (e.g.
      SDBus or NVMLDevicePool)
- [x] Non-exempt methods with logic branches have multiple
      tests
- [x] Mock objects derived from class have methods added
- [x] New public methods have doxygen comments describing
      behavior, inputs, outputs and exceptions raised

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Relates to #1973
- Fixes #1988

### CODE REVIEW

File: service/src/POSIXSignal.hpp [hunk 1]
```diff
@@ -78,6 +78,13 @@ namespace geopm
             /// @return m_info_s reduced information set from siginfo_t struct
             virtual m_info_s reduce_info(const siginfo_t &info) const = 0;
 
+            // @brief Use stat() to get the get the uid of /proc/<PID>
+            // virtual int pid_to_uid(int pid);
+
+            // @brief Use stat() to get the get the gid of /proc/<PID>
+            // virtual int pid_to_gid(int pid);
+
+
```
Comment (cmcantalupo) [inline service/src/POSIXSignal.hpp]:
```suggestion
```

File: service/src/POSIXSignal.hpp [hunk 2]
```diff
@@ -133,15 +135,16 @@ namespace geopm
             ///
             /// @remark See documentation for sigprocmask(2) about parameters
             ///         and return value.
-            virtual void sig_proc_mask(int how, const sigset_t *set, sigset_t *oldset) const = 0;
+            virtual void sig_proc_mask(int how, const sigset_t *sigset, sigset_t *oldset) const = 0;
```
Comment (Krebrov001) [inline service/src/POSIXSignal.hpp]:
```cpp
virtual void sig_proc_mask(int how, const sigset_t *sigset,
                           sigset_t *oldset) const = 0;
```

File: service/test/ServiceIOGroupTest.cpp [hunk 3]
```diff
@@ -37,6 +37,7 @@
 #include "geopm_topo.h"
 
 #include "geopm/Helper.hpp"
+#include "geopm/PlatformIO.hpp"
```
Comment (Krebrov001) [inline service/test/ServiceIOGroupTest.cpp:40]: Why do we need this header for, there are no other changes to the file which would indicate need to include this header in the file.

**Review by Krebrov001 — CHANGES_REQUESTED** [review]
Please implement these changes.

**Review by Krebrov001 — APPROVED** [review]
cool thanks

**Pull request merged**